addons/mission/functions/fnc_contaminationGas.sqf

```diff
@@ -2,22 +2,35 @@
 /*
  * Author: Mike
  * Adds contamination gas within a marker radius.
+ * Array returned can be used to clear the gas.
+ * Cloud Size cannot be greater than 3.
  *
  * Call from initPlayerLocal.sqf.
  *
  * Arguments:
  * 0: Marker <STRING>
  * 1: Colour RGBA <ARRAY> (default: [1, 1, 0, 0.06])
+ * 2: Gas cloud size <NUMBER> (default: 1)
  *
  * Return Value:
- * None
+ * Particle Emitters <ARRAY>
  *
  * Example:
  * ["MyMarker"] call MFUNC(contaminationGas)
- * ["MyMarker", [1, 1, 1, 0.04]] call MFUNC(contaminationGas)
+ * GVAR(gasOne) = ["MyMarker", [1, 1, 1, 0.04]] call MFUNC(contaminationGas)
  */
 
-params ["_marker", ["_colour", [1, 1, 0, 0.06]]];
+params ["_marker", ["_colour", [1, 1, 0, 0.06]], ["_cloudSize", 1]];
+
+// Marker shape should always be either "RECTANGLE" or "ELLIPSE"
+if (markerShape _marker == "") exitWith {
+    WARNING_1("Marker (%1) does not exist",_marker);
+};
+
+// Limit cloud size to 3 or less.
+if (_cloudSize > 3) exitWith {
+    WARNING_1("Cloud Size (%1) should be 3 or less.",_cloudSize);
```

Comment:
This isn't a "hard limit". You should actually set `_cloudSize = 3` in this case and also add to the warning message that it was limited to 3.